src/lib/Libifl/pbs_ifl_wrap.c

```diff
@@ -139,12 +139,6 @@
 # pragma warning disable 592
 #endif
 
-
-#if defined(__GNUC__) && defined(_WIN32) && !defined(SWIG_PYTHON_NO_HYPOT_WORKAROUND)
-/* Workaround for '::hypot' has not been declared', see https://bugs.python.org/issue11566 */
-# include <math.h>
-#endif
-
```

Comment:
What's the motivation for removing this? I'm worried that  it might break something in our hooks feature. If it's because it matches WIN32, but windows mom and pbs_python still use Python...

Reply:
If I go through the issue https://bugs.python.org/issue11566 :
- Windows affect version is not mentioned as 3.6
- without adding this didn't face any compiler error
- ran execjob_epilogoe and execjob_prologue hook without any error
- apparently looks like issue was for "the Microsoft Visual C++ Compiler for Python 2.7."

Please let me know whether I should test few more things after removing that header ?

Reply:
So I looked further into this file, src/lib/Libifl/pbs_ifl_wrap.c, and the heading says:
/* ----------------------------------------------------------------------------
 * This file was automatically generated by SWIG (http://www.swig.org).
 * Version 4.0.1
 *
 * This file is not intended to be easily readable and contains a number of
 * coding conventions designed to improve portability and efficiency. Do not make
 * changes to this file unless you know what you are doing--modify the SWIG
 * interface file instead.
 * ----------------------------------------------------------------------------- */

This file is only used on Windows, when 'swig' is not available to create pbs_ifl_wrap.c on the fly. Otherwise, it is auto-created via make (and WIndows gen_ifl_wrap.bat script) as follows:
bayucan@pbi-18-s12:~/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/src/lib/Libpython> rm -i pbs_ifl_wrap.c
rm: remove regular file ‘pbs_ifl_wrap.c’? y
bayucan@pbi-18-s12:~/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/src/lib/Libpython> make
echo '%module pbs_ifl' > pbs_ifl.i ; \
echo '%{' >> pbs_ifl.i ; \
echo '#include "pbs_ifl.h"' >> pbs_ifl.i ; \
echo '%}' >> pbs_ifl.i ; \
echo '%include "pbs_ifl.h"' >> pbs_ifl.i ; \
/home/bayucan/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/swig/bin/swig -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/swig/share/swig/3.0.10 -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/swig/share/swig/3.0.10/python \
        -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/openpbs/src/include \
        -python pbs_ifl.i
gcc -DHAVE_CONFIG_H -I. -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/openpbs/src/lib/Libpython -I../../../src/include  -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/openpbs/src/include -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/python/include/python3.6m -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/python/include/python3.6m  -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/tcltk/include -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/alsdk/include  -g -Wall -Werror -Wno-unused-result -fPIC -D_LARGEFILE_SOURCE -O2     -MT libpbspython_a-pbs_ifl_wrap.o -MD -MP -MF .deps/libpbspython_a-pbs_ifl_wrap.Tpo -c -o libpbspython_a-pbs_ifl_wrap.o `test -f 'pbs_ifl_wrap.c' || echo '/home/bayucan/2021.1.0/2021.0.0.20200728140711/openpbs/src/lib/Libpython/'`pbs_ifl_wrap.c
mv -f .deps/libpbspython_a-pbs_ifl_wrap.Tpo .deps/libpbspython_a-pbs_ifl_wrap.Po
rm -f libpbspython.a
ar cru libpbspython.a libpbspython_a-shared_python_utils.o libpbspython_a-common_python_utils.o libpbspython_a-pbs_ifl_wrap.o libpbspython_a-pbs_python_external.o libpbspython_a-pbs_python_svr_external.o libpbspython_a-module_pbs_v1.o libpbspython_a-pbs_python_svr_internal.o libpbspython_a-pbs_python_svr_size_type.o libpbspython_a-pbs_python_import_types.o 
ranlib libpbspython.a
gcc -DHAVE_CONFIG_H -I. -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/openpbs/src/lib/Libpython -I../../../src/include  -DLIBPYTHONSVR -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/openpbs/src/include -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/python/include/python3.6m -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/python/include/python3.6m  -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/tcltk/include -I/home/bayucan/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/alsdk/include  -g -Wall -Werror -Wno-unused-result -fPIC -D_LARGEFILE_SOURCE -O2     -MT libpbspython_svr_a-pbs_ifl_wrap.o -MD -MP -MF .deps/libpbspython_svr_a-pbs_ifl_wrap.Tpo -c -o libpbspython_svr_a-pbs_ifl_wrap.o `test -f 'pbs_ifl_wrap.c' || echo '/home/bayucan/2021.1.0/2021.0.0.20200728140711/openpbs/src/lib/Libpython/'`pbs_ifl_wrap.c
mv -f .deps/libpbspython_svr_a-pbs_ifl_wrap.Tpo .deps/libpbspython_svr_a-pbs_ifl_wrap.Po
rm -f libpbspython_svr.a
ar cru libpbspython_svr.a libpbspython_svr_a-shared_python_utils.o libpbspython_svr_a-common_python_utils.o libpbspython_svr_a-pbs_ifl_wrap.o libpbspython_svr_a-pbs_python_external.o libpbspython_svr_a-pbs_python_svr_external.o libpbspython_svr_a-module_pbs_v1.o libpbspython_svr_a-pbs_python_svr_internal.o libpbspython_svr_a-pbs_python_svr_size_type.o libpbspython_svr_a-pbs_python_import_types.o 
ranlib libpbspython_svr.a
bayucan@pbi-18-s12:~/2021.1.0/2021.0.0.20200728140711/linux312_x86_64-work/src/lib/Libpython> ls -l pbs_ifl_wrap.c
-rw-r--r-- 1 bayucan agt 327924 Jul 29 10:22 pbs_ifl_wrap.c

---
So I believe you don't need to update this file.